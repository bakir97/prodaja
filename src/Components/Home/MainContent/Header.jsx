import React, { Component } from "react";
import filtersFunction from "./filtersFunction";
import SearchBarComponent from "../SearchBar";
import Switch from "../Switch";
import LayoutButtons from "../LayoutButtons";
import Filters from "../Filters";
import Grid from "@material-ui/core/Grid";
export default class Header extends Component {
  state = {
    filters: false,
    filter: {
      minCijena: "",
      maxCijena: "",
      minTezina: "",
      maxTezina: ""
    }
  };
  filters = filter => {
    this.setState(filtersFunction(filter));
  };
  changeFiltersState = () => {
    this.setState(prevState => ({
      filters: !prevState.filters
    }));
  };
  search = e => {
    const { minCijena, minTezina, maxCijena, maxTezina } = this.state;
    const podaci = {};
    podaci.search = e;
    podaci.minCijena = minCijena;
    podaci.maxCijena = maxCijena;
    podaci.minTezina = minTezina;
    podaci.maxTezina = maxTezina;
    this.props.getProizvodi(podaci);
  };
  render() {
    const { width, changeLayout } = this.props;
    const { filters, minCijena, maxCijena, maxTezina, minTezina } = this.state;

    return (
      <>
        <Grid container justify="center">
          <Grid item container justify="center" xs={12}>
            <SearchBarComponent search={this.search} />
          </Grid>

          <Switch
            filters={filters}
            changeFiltersState={this.changeFiltersState}
          />
          {filters && (
            <Filters
              minCijena={minCijena}
              maxCijena={maxCijena}
              minTezina={minTezina}
              maxTezina={maxTezina}
              filters={this.filters}
            />
          )}
        </Grid>
        {width === "xs" ? null : <LayoutButtons changeLayout={changeLayout} />}
      </>
    );
  }
}
