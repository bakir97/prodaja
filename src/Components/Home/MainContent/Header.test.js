import Header from "./Header";
describe("Header", () => {
  const getProizvodi = jest.fn();
  const wrapper = shallow(<Header getProizvodi={getProizvodi} />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should call search", () => {
    wrapper.instance().search({});
    expect(getProizvodi).toBeCalled();
  });
});
