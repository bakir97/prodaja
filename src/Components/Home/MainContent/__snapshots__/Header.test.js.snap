// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<Fragment>
  <WithStyles(Grid)
    container={true}
    justify="center"
  >
    <WithStyles(Grid)
      container={true}
      item={true}
      justify="center"
      xs={12}
    >
      <SearchBarComponent
        search={[Function]}
      />
    </WithStyles(Grid)>
    <SwitchComponent
      changeFiltersState={[Function]}
      filters={false}
    />
  </WithStyles(Grid)>
  <LayoutButtons />
</Fragment>
`;
